pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -348,22 +347,18 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 		return err
 	}
 	// Coerce the CNI result version
-	res, err := cnitypes020.GetResult(resT)
+	res, err := cnitypescurrent.GetResult(resT)
 	if err != nil {
 		return fmt.Errorf("unable to understand network config: %v", err)
 	}
-	//TODO: v1.16 (khenidak) update NET_CONFIG_TEMPLATE to CNI version 0.3.0 or later so
-	// that we get multiple IP addresses in the returned Result structure
-	if res.IP4 != nil {
-		ipv4 = res.IP4.IP.IP.To4()
-	}
 
-	if res.IP6 != nil {
-		ipv6 = res.IP6.IP.IP
+	for _, ip := range res.IPs {
+		// Add the IP to tracked IPs
+		plugin.addPodIP(id, ip.Address.IP.String())
```

Comment:
@johscheuer 'res.IPs' might include addresses from interfaces *other* than the container interface. For each IP in the list, look at `ip.Interface` to find the index into the res.Interfaces list. If the item in the res.Interfaces list has a `sandbox != ""` then it's the container interface and you can use the IP address.

```
	for _, ipc := range res.IPs {
		if ipc.Interface == nil {
			continue
		}
		intIdx := *ipc.Interface
		if intIdx < 0 || intIdx >= len(res.Interfaces) || res.Interfaces[intIdx].Sandbox == "" {
			continue
		}
		<the IP address is a container/sandbox IP so use it>
```

Reply:
I just saw that https://github.com/kubernetes/kubernetes/pull/69821/files#diff-9768e7cd7bcf561500b1048edb02db5aR353-R362 has a similar method/function. Should this method be used here (as soon as it is merged)? Also I'm wondering why the checks are slightly different ?

Reply:
@johscheuer the logic is the same, though the code is slightly different. I think I prefer https://github.com/kubernetes/kubernetes/pull/69821/files#diff-9768e7cd7bcf561500b1048edb02db5aR353-R362 so just use that one.

Reply:
@dcbw, my suggestion is to move that helper function to some common place.

Reply:
@johscheuer still need to update the code above to handle https://github.com/kubernetes/kubernetes/pull/82844/files#r329772998